test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:test/test.dart';
+
+void main() {
+  Uri? tempDirUri;
+  final projectName = 'test_project';
+
+  setUp(() async {
+    tempDirUri = (await Directory.current.createTemp('.temp_test_')).uri;
+  });
+
+  tearDown(() async {
+    final dir = Directory(tempDirUri!.toFilePath());
+    if (await dir.exists()) {
+      await dir.delete(recursive: true);
+    }
+  });
+
+  test('Run Flutter', () async {
+    final projectDirUri = tempDirUri!.resolve('$projectName/');
+    final libDirUri = projectDirUri.resolve('lib/');
+    final bindingsGeneratedUri =
+        libDirUri.resolve('${projectName}_bindings_generated.dart');
+    final bindingsGeneratedCopyUri =
+        libDirUri.resolve('${projectName}_bindings_generated_copy.dart');
+
+    await Task.serial([
```

Comment:
All this Task stuff seems like a lot of boilerplate. Why not just write a `runProcess` and `copyFile` function, and do:

```dart
await runProcess(...);
await copyFile(...);
await runProcess(...);
```

Reply:
I had this laying around in another project. I can simplify it for this test if you prefer.

Reply:
Feel free to merge as-is if you're too busy to clean it up. It's only a test after all. I've already approved.